application/Modules/Mouse/Producers/MouseScrollEventProducer.cs

```diff
@@ -1,16 +1,36 @@
-﻿using System.ComponentModel.Composition;
+using System.Windows;
 using MORR.Modules.Mouse.Events;
 using MORR.Shared.Events.Queue;
+using MORR.Shared.Utility;
 
 namespace MORR.Modules.Mouse.Producers
 {
     /// <summary>
     ///     Provides a single-writer-multiple-reader queue for MouseScrollEvent
     /// </summary>
-    [Export(typeof(MouseScrollEventProducer))]
-    [Export(typeof(IReadOnlyEventQueue<MouseScrollEvent>))]
     public class MouseScrollEventProducer : DefaultEventQueue<MouseScrollEvent>
     {
-        // TODO: Implement this
+        public void StartCapture()
+        {
+            GlobalHook.AddListener(MouseHookCallback, NativeMethods.MessageType.WM_MOUSEWHEEL);
+            GlobalHook.IsActive = true;
+        }
+
+        public void StopCapture()
+        {
+            GlobalHook.RemoveListener(MouseHookCallback, NativeMethods.MessageType.WM_MOUSEWHEEL);
+        }
+
+        private void MouseHookCallback(GlobalHook.HookMessage hookMessage)
+        {
+            //One wheel click is defined as WHEEL_DELTA, which is 120. 
+            //(short)(((long)hookMessage.wParam >> 16) & 0xffff) retrieves the scrollAmount from wParam
+            var scrollAmount = (short) (((long) hookMessage.wParam >> 16) & 0xffff);
```

Comment:
The comment above doesn't really help explain what is happening - perhaps have an additional variable (note I also don't think the cast to long is correct, the parameter should be a DWORD, which is 32bit, not 64bit like `long`; which doesn't make a difference for the shift operation, but still)
```
var highOrderWord = (hookMessage.wParam.ToInt32() >> 16) & 0xffff;
var scrollAmount = (short) highOrderWord;
```

Reply:
Your suggestion is incorrect, the parameter in your context has already been modified in the global hook library and Karl and I have tried a long time to get the right ScrollAmount and the only functional code is var scrollAmount = (short) (((long) hookMessage.wParam >> 16) & 0xffff);

Reply:
Your suggestion causes overflow.

Reply:
System.OverflowException: 'Arithmetic operation resulted in an overflow.'

Reply:
Good catch - turns out I was wrong about it being 32bit. The following code should work just fine though:

```
var highOrderWord = (hookMessage.wParam.ToInt64() >> 16) & 0xffff;
var scrollAmount = (short) highOrderWord;
```